crates/kubelet/src/node/mod.rs

```diff
@@ -440,8 +436,8 @@ fn lease_spec_definition(node_name: &str) -> serde_json::Value {
 /// Default values and passed node-labels arguments are injected by config.
 fn node_labels_definition(arch: &str, config: &Config, builder: &mut Builder) {
     // Add mandatory static labels
-    builder.add_label("beta.kubernetes.io/os", "linux");
-    builder.add_label("kubernetes.io/os", "linux");
+    builder.add_label("beta.kubernetes.io/os", arch);
+    builder.add_label("kubernetes.io/os", arch);
```

Comment:
This appears to resolve the `kube-proxy` issue (at least on Kind which uses a node selector on this label). It isn't clear to me if this field as actually intended to indicate the host OS, or the *type* of container that it can run (Windows vs. Linux). Kind appears to be using it for the latter which tells me that we should not be saying `linux` (we shouldn't hardcode this anyway given that we target other systems). Thoughts?

Reply:
Honestly I am not sure of this myself. Gonna have to do some digging around to see what it is supposed to be so we are not bastardizing it too much

Reply:
Looks like this label is set to `runtime.GOOS` in the normal Kubelet: https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#kubernetes-io-os. However, I think because this solves our problem, we abuse it for our own. This will require us to document that though

Reply:
I think the issue arises from the host and workload "OS" not matching. Since we are specifying the arch differently from the host arch, I think this is fine. Perhaps we can leave it unset? That would be more accurate I think.

Reply:
Leaving it unset would be a breaking change wouldn't it? We'll just have to advertise that and test to make sure there aren't any problems